sigopt/xgboost/run.py

```diff
@@ -278,10 +278,15 @@ def train_xgb(self):
 
   def log_validation_metrics(self):
     # Always log xgb-default eval_metric
+    n_eval_rounds = 0
     if self.evals_result is not None:
       for dataset, metric_dict in self.evals_result.items():
         for metric_label, metric_record in metric_dict.items():
           self.run.log_metric(f"{dataset}-{metric_label}", metric_record[-1])
+          n_eval_rounds = len(metric_record)
+
+      if self.early_stopping_rounds:
+        self.run.log_metric('num_boost_round_before_stopping', n_eval_rounds)
```

Comment:
You can get this much easier with just the `booster.best_iteration` call. Which should probably be what we call it as well.

Reply:
I think the best_iteration is not what we want, if the loss / eval_metric drops later, then best_iteration won't match num_boost_round_before_stopping

Reply:
Am I misunderstanding this from the xgboost doc? 
https://xgboost.readthedocs.io/en/stable/python/python_intro.html#early-stopping
```
 If early stopping occurs, the model will have three additional fields: bst.best_score, bst.best_iteration.
```

Reply:
The docs say: "_Note that xgboost will return a model from the last iteration, not the best one_", so I think it's implied that best_iteration is not necessarily the final one (which is what num_boost_round_before_stopping records)

Reply:
Ok I see. It's probably something like 
```
num_boost_round_before_stopping - best_iteartion <= early_stopping_round
```
We should record both `best_iteration` and `num_boost_round_before_stopping`.  I did find this, which I think is equivalent to your `nbs before stopping`.
https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.Booster.num_boosted_rounds